geode-core/src/main/java/org/apache/geode/distributed/internal/InternalDistributedSystem.java

```diff
@@ -2464,6 +2464,10 @@ private void reconnect(boolean forcedDisconnect, String reason) {
     configProps.putAll(this.config.toSecurityProperties());
 
     int timeOut = oldConfig.getMaxWaitTimeForReconnect();
+    int memberTimeout = oldConfig.getMemberTimeout();
+    final int intervalsAllowedForFailureDetection = 4;
```

Comment:
Is there any significance to picking 4 intervals here?  Or was this just picked empirically because it was resulted in long enough timeouts to avoid the issue?  I assume since member-timeout and max-wait-time-reconnect are configurable by a user that they could circumvent this fix by setting both of those to arbitrarily low.  Is that a fair statement?

Reply:
I'm adding a comment to explain that setting:
    // we need to make sure that a surviving member is able
    // to take over coordination before trying to auto-reconnect.
    // failure detection can take 4 member-timeout intervals
    // so we set that as a minimum. (suspect, check suspect, final check, send new view)